### PR TITLE
Enable CacheManager and CachingProvider config to be updated on restore

### PR DESCRIPTION
**Pull request by anjumfatima90**
#26845

### CODE REVIEW

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

File: dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/Activator.java [hunk 1]
```diff
@@ -82,7 +85,8 @@ public void removedService(ServiceReference<Condition> ref, Boolean present) {
 
     @Override
     public void configurationEvent(ConfigurationEvent event) {
-        if ("io.openliberty.jcache.cachemanager".equals(event.getFactoryPid())
+        Set<String> pids = new HashSet<>(Arrays.asList("io.openliberty.jcache.cachemanager", "io.openliberty.jcache.cachingprovider"));
```
Comment (tjwatson) [inline dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/Activator.java]: This should be a static final `Set<String>`

File: dev/io.openliberty.checkpoint_fat_session_cache_infinispan/fat/src/io/openliberty/checkpoint/session/cache/infinispan/CheckpointSessionCacheOneServerTest.java [hunk 2]
```diff
@@ -89,14 +92,17 @@ public static void setUp() throws Exception {
         options.put("-Dinfinispan.cluster.name", rand);
         options.put("-Dsession.cache.config.file", sessionCacheConfigFile);
         server.setJvmOptions(options);
-        server.setCheckpoint(CheckpointPhase.AFTER_APP_START, true, null);
+        server.setCheckpoint(CheckpointPhase.AFTER_APP_START, false, null);
         server.startServer();
+        configureEnvVariable(server, singletonMap("uri_location", "file:///${shared.resource.dir}/infinispan/infinispan-attr-meta-configuration.xml"));
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat_session_cache_infinispan/fat/src/io/openliberty/checkpoint/session/cache/infinispan/CheckpointSessionCacheOneServerTest.java:97]: I am surprised variable substitution can be done in a value from the environment.  So this results in the final value for the URI to have filled in the shared resource dir?

File: dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/Activator.java (hunk 1 above)
Comment (anjumfatima90) [reply dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/Activator.java]: made it final, thanks!

File: dev/io.openliberty.checkpoint_fat_session_cache_infinispan/fat/src/io/openliberty/checkpoint/session/cache/infinispan/CheckpointSessionCacheOneServerTest.java (hunk 2 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat_session_cache_infinispan/fat/src/io/openliberty/checkpoint/session/cache/infinispan/CheckpointSessionCacheOneServerTest.java:97]: Yes, the final value has the shared resource dir location.

File: dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/Activator.java (hunk 1 above)
Comment (tjwatson) [reply dev/io.openliberty.jcache.internal/src/io/openliberty/jcache/internal/Activator.java]: But it should also be a static constant.  The final var is currently re-created each call to `configureEvent`.

**Review by tjwatson — APPROVED** [review]

**Comment by anjumfatima90** [comment]
#libby

**Pull request merged**